machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,389 @@
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MCOConfig `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ControllerConfig describes configuration for MachineConfigController.
+// This is currently only used to drive the machineconfigs generated by the TemplateController.
+type ControllerConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ControllerConfigSpec   `json:"spec"`
+	Status ControllerConfigStatus `json:"status"`
+}
+
+// ControllerConfigSpec is the spec for ControllerConfig resource.
+type ControllerConfigSpec struct {
+	ClusterDNSIP        string `json:"clusterDNSIP"`
+	CloudProviderConfig string `json:"cloudProviderConfig"`
+
+	// The openshift platform, e.g. "libvirt", "openstack", "aws", or "none"
+	Platform string `json:"platform"`
+
+	EtcdDiscoveryDomain string `json:"etcdDiscoveryDomain"`
+
+	// CAs
+	EtcdCAData       []byte `json:"etcdCAData"`
+	EtcdMetricCAData []byte `json:"etcdMetricCAData"`
+	RootCAData       []byte `json:"rootCAData"`
+
+	// PullSecret is the default pull secret that needs to be installed
+	// on all machines.
+	PullSecret *corev1.ObjectReference `json:"pullSecret,omitempty"`
+
+	// Images is map of images that are used by the controller.
+	Images map[string]string `json:"images"`
+
+	// Sourced from configmap/machine-config-osimageurl
+	OSImageURL string `json:"osImageURL"`
+}
+
+// ControllerConfigStatus is the status for ControllerConfig
+type ControllerConfigStatus struct {
+	// The generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Represents the latest available observations of current state.
+	// +optional
+	Conditions []ControllerConfigStatusCondition `json:"conditions,omitempty"`
+}
+
+// ControllerConfigStatusCondition contains condition information for ControllerConfigStatus
+type ControllerConfigStatusCondition struct {
+	// type specifies the state of the operator's reconciliation functionality.
+	Type ControllerConfigStatusConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status object.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// reason is the reason for the condition's last transition.  Reasons are CamelCase
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	Message string `json:"message,omitempty"`
+}
+
+// ControllerConfigStatusConditionType valid conditions of a machineconfigpool
+type ControllerConfigStatusConditionType string
+
+const (
+	TemplateContollerRunning   ControllerConfigStatusConditionType = "TemplateContollerRunning"
```

Comment:
the `TemplateContoller` prefix is necessary?

Also note the missing `r`.

Reply:
Yeah (predates our team, cc @abhinavdahiya )

Reply:
(just realized is a typoooo as well)